CMake/External_VTK.cmake

```diff
@@ -54,13 +54,31 @@ set(vtk_cmake_args ${vtk_cmake_args}
   )
 
 # Qt
-add_package_dependency(
-  PACKAGE VTK
-  PACKAGE_DEPENDENCY Qt
-  PACKAGE_DEPENDENCY_ALIAS Qt4
-)
-if(QT_QMAKE_EXECUTABLE)
-  set(BUILD_QT_WEBKIT ${QT_QTWEBKIT_FOUND})
+if(Qt_version_major EQUAL 4)
+  add_package_dependency(
+    PACKAGE VTK
+    PACKAGE_DEPENDENCY Qt
+    PACKAGE_DEPENDENCY_ALIAS Qt4
+  )
+  if(QT_QMAKE_EXECUTABLE)
+    set(BUILD_QT_WEBKIT ${QT_QTWEBKIT_FOUND})
+    set(vtk_cmake_args ${vtk_cmake_args}
+      -DVTK_RENDERING_BACKEND:STRING=OpenGL2
+      -DQT_QMAKE_EXECUTABLE:PATH=${QT_QMAKE_EXECUTABLE}
+    )
+  endif()
+else()
+  add_package_dependency(
+    PACKAGE VTK
+    PACKAGE_DEPENDENCY Qt
+    PACKAGE_DEPENDENCY_ALIAS Qt5
+    PACKAGE_DEPENDENCY_COMPONENTS
+      Core Gui Widgets OpenGL Designer UiPlugin
```

Comment:
Could you please add an if check for:

```cmake
if (Qt5OpenGL_DIR)
  Module_vtkGUISupportQtOpenGL:BOOL=ON
endif
```

Reply:
Does `find_package` succeed if not all components are found? I didn't think it did, but I guess it depends on the find module. Anyway, no OpenGL should be a hard error; fletch building a VTK that can't be used by our GUI's rather defeats the purpose.

Reply:
I don't think there would be many users for the vtkGUISupportQtOpenGL module. I know MAP-Tk doesn't use it. That should be an optional module.

Reply:
> IMNSHO, Qt5 without OpenGL is not very useful;

Care to explain..

Reply:
For one, because much of Qt is itself implemented with OpenGL these days. For two, because it's just one of the things I expect to be present in a Qt installation.

What is the problem, anyway? Do you actually have a Qt build that does *not* have OpenGL?

Reply:
FWIW, our current VTK configuration assumes OpenGL enabled Qt. I am in favor of keeping it that way.

Reply:
No problem, was just curious why that module is required.